### PR TITLE
HBASE-24572 release scripts should try to use a keyid when refering to GPG keys.

### PR DESCRIPTION
**Pull request by busbey**
first pass at turning a given gpg key name into a key id during release info gathering.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   1m 54s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2001 |
| Optional Tests |  |
| uname | Linux c39ec8edf31f 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 982bd5fadd |
| Max. process+thread count | 57 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 28s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m 53s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2001 |
| Optional Tests |  |
| uname | Linux 65e9e42684ee 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 982bd5fadd |
| Max. process+thread count | 47 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  shellcheck  |   0m  1s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   3m  3s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2001 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux fe03474b142c 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 982bd5fadd |
| Max. process+thread count | 44 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by ndimiduk** [comment]
So it makes a guess, but that guess is not correct in my case. I want it to use the dedicated signing key (`0xAD9039071C3489BD`), not the key of the master keyring.

```
$ gpg -K
/Users/ndimiduk/.gnupg/pubring.kbx
----------------------------------
sec#  rsa4096/0xCA3633F18644EEB6 2014-03-11 [SC] [expires: 2024-06-12]
      Key fingerprint = 3A74 917C 0C45 844F B816  BB4A CA36 33F1 8644 EEB6
uid                   [ultimate] Nick Dimiduk <ndimiduk@apache.org>
uid                   [ultimate] Nick Dimiduk <ndimiduk@gmail.com>
ssb#  rsa4096/0xB44B40F2D2DCE494 2014-03-11 [E] [expires: 2022-06-12]
ssb>  rsa4096/0xAD9039071C3489BD 2015-04-30 [S] [expires: 2022-06-12]
ssb>  rsa4096/0x2FD7719BB28DF80C 2020-06-15 [A] [expires: 2021-06-15]
$ ./dev-support/create-release/do-release-docker.sh -d ~/build-rc2-1 -f
========================
=== Gathering release details.
...
GPG_KEY [ndimiduk@apache.org]: 
We think the key 'ndimiduk@apache.org' corresponds to the key id '0x1C3489BD'. Is this correct [y/n? 
```

File: dev-support/create-release/release-util.sh [hunk 1]
```diff
@@ -236,6 +236,14 @@ function get_release_info {
 
   GIT_EMAIL="$ASF_USERNAME@apache.org"
   GPG_KEY="$(read_config "GPG_KEY" "$GIT_EMAIL")"
+  if ! GPG_KEY_ID=$("${GPG}" "${GPG_ARGS[@]}" --keyid-format 0xshort --list-public-key "${GPG_KEY}" | grep "\[S\]" | grep -o "0x[0-9A-F]*") ||
+      [ -z "${GPG_KEY_ID}" ] ; then
+    GPG_KEY_ID=$("${GPG}" "${GPG_ARGS[@]}" --keyid-format 0xshort --list-public-key "${GPG_KEY}" | head -n 1 | grep -o "0x[0-9A-F]*" || true)
+  fi
+  read -r -p "We think the key '${GPG_KEY}' corresponds to the key id '${GPG_KEY_ID}'. Is this correct [y/n? " ANSWER
```
Comment (ndimiduk) [inline dev-support/create-release/release-util.sh:243]: missing closing brace on `[y/n?`

**Comment by ndimiduk** [comment]
What am I saying, it has chosen the key I intended. `0xAD9039071C3489BD` is `0x1C3489BD` How does that work?

**Comment by busbey** [comment]
the former is the 0xlong key id and the latter is the 0xshort key id. if you gave the 0xlong key id in the `get_release_info` prompt then saying "n" to the guess will have the rest of the tooling use the version you gave. (and the confirmation prompt for the release info should reflect that)

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 37s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  8s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   1m 50s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2001 |
| Optional Tests |  |
| uname | Linux b5c3aa45e34c 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e614b89c33 |
| Max. process+thread count | 52 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/2/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   8m 17s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  6s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   9m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2001 |
| Optional Tests |  |
| uname | Linux 83beae451011 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e614b89c33 |
| Max. process+thread count | 47 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/2/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   8m 18s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  shellcheck  |   0m  2s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   9m 38s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2001 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 2136506ca208 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e614b89c33 |
| Max. process+thread count | 46 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/2/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  5s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  shelldocs  |   0m  0s |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  6s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  shellcheck  |   0m  2s |  There were no new shellcheck issues.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +0 :ok: |  asflicense  |   0m  0s |  ASF License check generated no output?  |
|  |   |   2m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2001 |
| Optional Tests | dupname asflicense shellcheck shelldocs |
| uname | Linux 82e7d23c65d6 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3bd54217a0 |
| Max. process+thread count | 47 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  3s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  6s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   1m 55s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2001 |
| Optional Tests |  |
| uname | Linux d2cf28e6d017 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3bd54217a0 |
| Max. process+thread count | 43 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  6s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  8s |  Maven dependency ordering for branch  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
||| _ Other Tests _ |
|  |   |   2m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.12 Server=19.03.12 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2001 |
| Optional Tests |  |
| uname | Linux 7865e6fa173e 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 3bd54217a0 |
| Max. process+thread count | 46 (vs. ulimit of 12500) |
| modules | C:  U:  |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-2001/1/console |
| versions | git=2.17.1 maven=(cecedd343002696d0abb50b32b541b8a6ba2883f) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**